libsweep/src/sweep.cc

```diff
@@ -288,111 +235,72 @@ void sweep_device_accumulate_scans(sweep_device_s device) {
       received = 1;
     }
   }
+} catch (const std::exception& e) {
+  // worker thread is dead at this point
```

Comment:
Ahh that makes more sense. Would it be better to keep the queue abstract and put the data member in the device structure? Or does it seem logical to put it in the queue given this is all related to thread communication... and the queue is designed as a thread safe queue.